node/src/test_utils/persistent_configuration_mock.rs

```diff
@@ -44,6 +44,10 @@ pub struct PersistentConfigurationMock {
     mapping_protocol_results: RefCell<Vec<Result<Option<AutomapProtocol>, PersistentConfigError>>>,
     set_mapping_protocol_params: Arc<Mutex<Vec<Option<AutomapProtocol>>>>,
     set_mapping_protocol_results: RefCell<Vec<Result<(), PersistentConfigError>>>,
+    min_hops_count_params: Arc<Mutex<Vec<()>>>,
```

Comment:
I prefer it the other way, as it makes the tests easier to read. When a reader says two values in the params, and we're asserting on it, a user can be confident that the function ran exactly 2 times and we expect this behaviour.

Reply:
You may prefer it but we've historically fought against that... You'd find many and many mocks where the param assertions for certain methods simply miss. 

Your solution is unusual and it is so called non-standard mock. There were times, when I was early here, when Dan insisted on avoiding that by all means. I remember I simply learned to obey him in this matter as it made quite a lot of sense to me. Asserting this way looks awkward and you could hardly see Dan to decide for it. As I said in the previous comment, it would've had to be a special case.